src/common/io/socket/common.c

```diff
@@ -126,3 +128,86 @@ sckOptionSet(int fd)
 
     FUNCTION_TEST_RETURN_VOID();
 }
+
+/***********************************************************************************************************************************
+Use poll() to determine when data is ready to read/write on a socket. Retry after EINTR with whatever time is left on the timer.
+***********************************************************************************************************************************/
+// Helper to determine when poll() should be retried
+static bool
+sckReadyRetry(int pollResult, int errNo, bool first, Wait *wait)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(INT, pollResult);
+        FUNCTION_TEST_PARAM(INT, errNo);
+        FUNCTION_TEST_PARAM(BOOL, first);
+        FUNCTION_TEST_PARAM(WAIT, wait);
+    FUNCTION_TEST_END();
+
+    // No retry by default
+    bool result = false;
+
+    // Process errors
+    if (pollResult == -1)
+    {
+        // Don't error on an interrupt. If the interrupt lasts long enough there may be a timeout, though.
+        if (errNo != EINTR)
+            THROW_SYS_ERROR_CODE(errNo, KernelError, "unable to poll socket");
+
+        // Retry if first iteration or time remaining
+        result = first || waitMore(wait);
+    }
+
+    FUNCTION_TEST_RETURN(result);
+}
+
+bool
+sckReady(int fd, bool read, bool write, TimeMSec timeout)
```

Comment:
Worse, we use select() in two other places (and one of them does select() on multiple FDs). But I thought it was too much to try to unify them all to fix this issue.

Eventually they should all end up in `fd.c` or something because they are all the same thing.

Reply:
Should we maybe have the result checked as == 1..?  If poll returned something else that would mean there is some other issue..?

Reply:
poll() always returns -1 on error according to the man page. What else would we test for?